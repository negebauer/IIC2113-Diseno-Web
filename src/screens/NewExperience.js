import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { devlog } from '../utils/log'
import { createExperience } from '../redux/modules/experience'
import { replace } from '../redux/modules/router'

const mapStateToProps = state => ({
  loading: state.user.loading,
  api_key: state.user.api_key,
  error: state.user.error,
})

const mapDispatchToProps = {
  replace,
  createExperience,
}

class NewExperience extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      description: '',
      sentExperience: false,
    }
  }

  componentDidUpdate = () => {
    if (this.state.sentExperience) {
      this.props.replace('/experiences')
    }
  }

  submit = e => {
    e.preventDefault()
    devlog(this.state.mail)
    devlog(this.state.password)
    this.setState({
      sentExperience: true,
    })
    this.props.createExperience(this.state)
  }

  handleChange = ev =>
    this.setState({
      [ev.target.name]: Number(ev.target.value) || ev.target.value,
    })

  render() {
    devlog('Login', this.props)
    const message =
      (this.props.loading && 'Cargando') ||
      (this.props.error && this.props.error) ||
      'Nueva Experiencia'
    return (
      <div className="container">
        <div className="row">
          <form className="col s12" ref={f => (this.form = f)}>
            <h2 className="header">
              {message}
            </h2>
            <div className="row">
              <div className="input-field col s12">
                <input
                  id="name"
                  type="text"
                  className="validate"
                  name="name"
                  value={this.state.name}
                  onChange={this.handleChange}
                />
                <label htmlFor="name">Name</label>
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <input
                  id="description"
                  type="text"
                  className="validate"
                  name="description"
                  value={this.state.description}
                  onChange={this.handleChange}
                />
                <label htmlFor="description">Descripcion</label>
              </div>
            </div>
            <button
              className="waves-effect waves-light btn-large"
              onClick={this.submit}
            >
              Login
            </button>
          </form>
        </div>
      </div>
    )
  }
}

NewExperience.propTypes = {
  api_key: PropTypes.string.isRequired,
  error: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
  fetchExperiences: PropTypes.func.isRequired,
  replace: PropTypes.func.isRequired,
  createExperience: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(NewExperience)
