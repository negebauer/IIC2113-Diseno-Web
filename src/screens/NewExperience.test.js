import React from 'react'
import createHistory from 'history/createBrowserHistory'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
// import _ from 'lodash/fp/object'
import NewExperience from './NewExperience'
import configureStore from '../redux/store'

let history
let initialState
let store

beforeEach(() => {
  history = createHistory()
  initialState = {}
  const api = { createExperience: () => {} }
  store = configureStore(initialState, { history, api })
})

it('shallow renders without crashing', () => {
  shallow(<NewExperience store={store} />)
})

it('mount renders without crashing', () => {
  mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <NewExperience />
      </ConnectedRouter>
    </Provider>
  )
})

it('matches snapshot', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <NewExperience />
      </ConnectedRouter>
    </Provider>
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders children', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <NewExperience />
      </ConnectedRouter>
    </Provider>
  )
  expect(wrapper.children().exists()).toBeTruthy()
})
