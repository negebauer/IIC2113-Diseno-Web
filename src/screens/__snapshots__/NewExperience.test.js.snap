// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectedRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NewExperience)>
        <NewExperience
          api_key=""
          createExperience={[Function]}
          creating={false}
          error=""
          loading={false}
          replace={[Function]}
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <form
                className="col s12"
              >
                <h2
                  className="header"
                >
                  Nueva Experiencia
                </h2>
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <input
                      className="validate"
                      id="name"
                      name="name"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="name"
                    >
                      Name
                    </label>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <input
                      className="validate"
                      id="description"
                      name="description"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="description"
                    >
                      Descripcion
                    </label>
                  </div>
                </div>
                <button
                  className="waves-effect waves-light btn-large"
                  onClick={[Function]}
                >
                  Agregar experiencia
                </button>
              </form>
            </div>
          </div>
        </NewExperience>
      </Connect(NewExperience)>
    </Router>
  </ConnectedRouter>
</Provider>
`;
